### PR TITLE
Add color override classes

### PR DESCRIPTION
**Pull request by janfaracik**
This MR adds a series of color override classes, as well as the rest of the color spectrum for use on the frontend.

This means its now possible to use a color override class such as `jenkins-!-color-indigo` to set the color of text/symbols to `indigo`.

The follow colors have been added, including light and dark variations (prefix the color with `light-` or `dark-`, e.g. `var(--light-blue)`) :

```
  --blue: #007aff;
  --brown: #a2845e;
  --cyan: #32ade6;
  --green: #1c9146;
  --indigo: #5856d6;
  --orange: #fe820a;
  --pink: #fb0f45;
  --purple: #af52de;
  --red: #df001c;
  --yellow: #ffcc00;
```

There are now also additional semantic colors, based off of the colors above:

```
--error-color: var(--red);
--warning-color: var(--orange);
--success-color: var(--green);
--destructive-color: var(--red);
--build-color: var(--green);
```

Plugin developers can easily create their own semantic colors for use in their plugins (and remain consistent with Jenkins) by using the provided color palette.

---

I've also removed the table button color overrides as they're no longer needed, I've also removed the color.css file and moved it's contents to colors.less.

I'll update the Design Library to include the new colors as well as how to use the new color override classes.

### Proposed changelog entries

* Developer: New series of colours and new range of semantic colours added (see [Design Library](https://plugins.jenkins.io/design-library/) for more details).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Comment by uhafner** [comment]
Thanks for making more colors available!

Would it make sense to describe in the design library how and when to use these colors? E.g., when we show warnings, test results or coverage ranges: should we use `green` for success, `red` for errors, etc.? Or would it make more sense to add semantic classes as well that map to real colors?

Does it also make sense to have some lighter or darker versions (shading) for charts? Or should I better create them on my own in my plugins? I think at least some advice would be helpful!

**Comment by daniel-beck** [comment]
What is the purpose of color specifications that aren't coming from semantic meanings (e.g. destructive button = red)?

How would you expect color-restricted themes to define these colors (like Solarized)?

How would you expect colorblind themes to define these colors, and to reconcile multiple uses of the same color for different semantic actions?

**Comment by janfaracik** [comment]
> Would it make sense to describe in the design library how and when to use these colors? E.g., when we show warnings, test results or coverage ranges: should we use `green` for success, `red` for errors, etc.? Or would it make more sense to add semantic classes as well that map to real colors?

Absolutely! I've added some semantic colors now, although more will definitely be added:

```
--error-color: var(--red);
--warning-color: var(--orange);
--success-color: var(--green);
```

> Does it also make sense to have some lighter or darker versions (shading) for charts? Or should I better create them on my own in my plugins? I think at least some advice would be helpful!

I've added some light and dark variations :) I think its best that Jenkins provide those variations so that they're consistent across the system.

> What is the purpose of color specifications that aren't coming from semantic meanings (e.g. destructive button = red)?

The color specifications exist to provide a consistent set of colors that can be a basis for semantic colors, whether that be semantic colors that we've defined or ones that plugin developers have. 

> How would you expect color-restricted themes to define these colors (like Solarized, which has a handful of highlight colors but I think fewer than these)?

Solarized can always rely on the defaults if need be, or provide it's own muted set of colors.

> How would you expect colorblind themes to define these colors, and to reconcile multiple uses of the same color for different semantic actions?

I imagine a colorblind theme will primarily override the semantic versions of the colors rather than the base palette, although I don't have much experience in this area admittedly.

**Review by uhafner — APPROVED** [review]
I think this is a very good starting point! I'll try to see how this works in my plugins. If we need more semantic symbols or more colors we still can add them later in a subsequent PR.

**Review by NotMyFault — APPROVED** [review]
🎉

**Comment by janfaracik** [comment]
Working on docs at the moment, will update the PR once I've got something concrete 👍

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: war/src/main/less/abstracts/theme.less
Comment (basil) [inline war/src/main/less/abstracts/theme.less:324]: Seems to cause the linter to fail on Windows as described in [JENKINS-68887](https://issues.jenkins.io/browse/JENKINS-68887). Perhaps there should be a semicolon after the closing parenthesis?

**Pull request merged**
